api/rpc_test.go

```diff
@@ -110,6 +115,188 @@ func TestModulesImplementFullAPI(t *testing.T) {
 	}
 }
 
+// TestAuthedRPC tests whether a client with admin permissions can access all
+// methods at all permission levels via RPC.
+func TestAuthedRPC(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	// generate dummy signer and sign admin perms token with it
+	signer, err := jwt.NewHS256(make([]byte, 32))
+	require.NoError(t, err)
+	token, err := perms.NewTokenWithPerms(signer, perms.AllPerms)
+	require.NoError(t, err)
+
+	nd, server := setupNodeWithAuthedRPC(t, signer)
+	url := nd.RPCServer.ListenAddr()
+
+	// we need to run this a few times to prevent the race where the server is not yet started
+	var rpcClient *client.Client
+	require.NoError(t, err)
+	for i := 0; i < 3; i++ {
+		time.Sleep(time.Second * 1)
+		rpcClient, err = client.NewClient(ctx, "http://"+url, string(token))
```

Comment:
I know it is just a test, but do we plan to use `tls` to protect token from MITM for rpc?

Reply:
This seems to be a test where TLS is not needed, but the concern is totally valid and we `must` use TLS, so that we don't transfer access token over plaintext